packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx

```diff
@@ -165,31 +94,65 @@ export const Datepicker: ComponentWithAs<'div', DatepickerProps> &
   const context = useFluentContext();
   const { setStart, setEnd } = useTelemetry(Datepicker.displayName, context.telemetry);
   setStart();
+  const dateSelectionErrorString = 'A date selection is required';
   const datepickerRef = React.useRef<HTMLElement>();
-  const [open, setOpen] = React.useState<boolean>(false);
+  const [openState, setOpenState] = useAutoControlled<OpenState>({
+    defaultValue: props.defaultCalendarOpenState ? OpenState.Open : OpenState.Closed,
+    value: undefined,
+    initialValue: OpenState.Closed,
+  });
   const [selectedDate, setSelectedDate] = React.useState<Date | undefined>();
-  const valueFormatter = date => (date ? formatMonthDayYear(date, DEFAULT_LOCALIZED_STRINGS) : '');
-  const {
-    firstDayOfWeek,
-    firstWeekOfYear,
-    dateRangeType,
-    calendar,
-    popup,
-    className,
-    design,
-    styles,
-    variables,
-  } = props;
-  const calendarOptions: IDayGridOptions = {
-    selectedDate: selectedDate ?? props.today ?? new Date(),
-    navigatedDate: selectedDate ?? props.today ?? new Date(),
-    firstDayOfWeek,
-    firstWeekOfYear,
-    dateRangeType,
-  };
+  const [formattedDate, setFormattedDate] = React.useState<string>('');
+  const [error, setError] = React.useState<string>(() =>
+    props.isRequired && !selectedDate ? dateSelectionErrorString : '',
+  );
 
-  const showCalendarGrid = () => {
-    setOpen(true);
+  const { calendar, popup, className, design, styles, variables, formatMonthDayYear } = props;
+  const valueFormatter = date => (date ? formatMonthDayYear(date) : '');
```

Comment:
Please move out it from component's body

Reply:
@layershifter  in this same file or in different one?

Reply:
The same

Reply:
Should I move the `valueFormatter` away from component's body even though it depends on props?

Reply:
I see, now it depends on props. Ignore it